aws-redshift-eventsubscription/src/main/java/software/amazon/redshift/eventsubscription/Translator.java

```diff
@@ -183,15 +184,8 @@ static ResourceModel translateFromReadTagsResponse(final DescribeTagsResponse aw
     static ModifyTagsRequest translateToUpdateTagsRequest(final ResourceModel desiredResourceState,
                                                           final ResourceModel currentResourceState,
                                                           final String resourceName) {
-        List<Tag> toBeCreatedTags = desiredResourceState.getTags() == null ? Collections.emptyList() : desiredResourceState.getTags()
-                .stream()
-                .filter(tag -> currentResourceState.getTags() == null || !currentResourceState.getTags().contains(tag))
-                .collect(Collectors.toList());
-
-        List<Tag> toBeDeletedTags = currentResourceState.getTags() == null ? Collections.emptyList() : currentResourceState.getTags()
-                .stream()
-                .filter(tag -> desiredResourceState.getTags() == null || !desiredResourceState.getTags().contains(tag))
-                .collect(Collectors.toList());
+        List<Tag> toBeCreatedTags = subtract(desiredResourceState.getTags(), currentResourceState.getTags());
+        List<Tag> toBeDeletedTags = subtract(currentResourceState.getTags(), desiredResourceState.getTags());
```

Comment:
this implies set difference ?

Reply:
I implement `subtract` method to replace this part in order to reduce code duplicity. The fundamental logic is the same. I've already implemented this part in ClusterParameterGroup: [here](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-redshift/blob/master/aws-redshift-clusterparametergroup/src/main/java/software/amazon/redshift/clusterparametergroup/Translator.java#L285-L302) and it works for months.